qiskit_experiments/library/randomized_benchmarking/data/generate_clifford_data.py

```diff
@@ -0,0 +1,187 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+This file is a stand-alone script for generating the npz files in
+:mod:`~qiskit_experiment.library.randomized_benchmarking.clifford_utils.data` directory.
+
+The script relies on the values of ``_CLIFF_SINGLE_GATE_MAP_2Q``
+in :mod:`~qiskit_experiment.library.randomized_benchmarking.clifford_utils`
+so they must be set correctly before running the script.
+"""
+import itertools
+
+import numpy as np
+import scipy.sparse
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit.library import (
+    IGate,
+    HGate,
+    SXdgGate,
+    SGate,
+    XGate,
+    SXGate,
+    YGate,
+    ZGate,
+    SdgGate,
+    CXGate,
+    CZGate,
+)
+from qiskit.quantum_info.operators.symplectic import Clifford
+from qiskit_experiments.library.randomized_benchmarking.clifford_utils import (
+    CliffordUtils,
+    _CLIFF_SINGLE_GATE_MAP_1Q,
+    _CLIFF_SINGLE_GATE_MAP_2Q,
+)
+
+NUM_CLIFFORD_1Q = CliffordUtils.NUM_CLIFFORD_1_QUBIT
+NUM_CLIFFORD_2Q = CliffordUtils.NUM_CLIFFORD_2_QUBIT
+
+
+def _hash_cliff(cliff):
+    """Produce a hashable value that is unique for each different Clifford.  This should only be
+    used internally when the classes being hashed are under our control, because classes of this
+    type are mutable."""
+    table = cliff.table
```

Comment:
Good points, actually, I thought both of them.

Regarding `tableau`, it's quite new, i.e. introduced in the latest terra 0.22. If we use the property, experiments must require terra >= 0.22. I think it would be better to use `table` property until it is deprecated. But I know this code is only for developers (so far), so the requirement might not be so demanding (the requirement would live in `requirements-dev.txt`). Which do you like?

The code implementing hash originated from [Jake's PR in terra](https://github.com/Qiskit/qiskit-terra/blob/fdbd7516aac9938be5745cd326ca3b5b0d77afe5/qiskit/quantum_info/operators/symplectic/clifford.py#L756). I like the code because it should be faster than the general approach that uses string conversion.